evalml/automl/automl_search.py

```diff
@@ -313,13 +323,15 @@ def __init__(self,
             _, ensembling_indices, _, _ = split_data(X_shape, self.y_train, problem_type=self.problem_type, test_size=_ensembling_split_size, random_seed=self.random_seed)
             self.ensembling_indices = ensembling_indices.to_dataframe()[0].tolist()
 
-        self._engine = SequentialEngine(self.X_train,
-                                        self.y_train,
-                                        self.ensembling_indices,
-                                        self,
-                                        should_continue_callback=self._should_continue,
-                                        pre_evaluation_callback=self._pre_evaluation_callback,
-                                        post_evaluation_callback=self._post_evaluation_callback)
+        if not engine:
+            self._engine = SequentialEngine()
```

Comment:
@freddyaboulton @chukarsten : 

I am thinking that sooner or later we should switch this to default to `DaskEngine` with a local dask cluster. What would we need to do in order to make that happen? By my accounting: make sure the team feels good about this move, list dask in our requirements, update our documentation to explain what's going on and how to configure a remote cluster, and confirm that the perf testing we've done is adequate to make this change.

Reply:
I was kind of thinking the same thing working through all this.  It seems that, in the grand scheme of things, a DaskEngine with 1 worker eliminates the need for a SequentialEngine.  And yeah, that sounds about right, but the proper documentation for how to limit the DaskEngine to a certain number of workers and perhaps an expectation of how many default workers there should be are required. 

So currently the AutoMLSearch is designed to take in an `engine` argument.  The `EngineBase` object expects a `client` argument, in the case of a `DaskEngine`.  And the client object, when it's not run as a default instance, takes a `cluster` argument.  The `cluster` is where you lay out all the details behind the resources and amount of parallelism you want to use.  As such, I think we need to answer the questions 1.) what are the default settings for workers we want for the DaskEngine? and 2.) what are the default settings for workers we want for the AutoMLSearch?  We could either go the way of trying to make it as fast as possible for the user by default or protect their resource by default.  I think whichever of the two philosophies (faster vs. resource conservative) we choose, we should probably be consistent between DaskEngine and AutoMLSearch.